File: exs/mundo_1/javascript/020.js
/*
Desafio 020

Problema: O mesmo professor do desafio anterior quer sortear a ordem de 
          apresentação de trabalhos dos alunos. Faça um programa que leia o 
          nome dos quatro alunos e mostre a ordem sorteada

Resolução do problema:
*/

const input = require('readline-sync').question

let nome1 = input('1º aluno(a): ')
let nome2 = input('2º aluno(a): ')
let nome3 = input('3º aluno(a): ')
let nome4 = input('4º aluno(a): ')

let alunos = [
    nome1,
    nome2,
    nome3,
    nome4
]

let quantidadeAlunos = alunos.length // ou 4

// indice assume 0, 1, 2 ou 3 em ordem crescente
for (let indice = 0; indice < quantidadeAlunos; indice++) {
    // indiceAleatorio pode assumir 0, 1, 2 ou 3 sem ordem especifica 
    let indiceAleatorio = Math.trunc(Math.random() * quantidadeAlunos)

    // copia o nome de um aluno 
    let alunoAleatorio = alunos[indiceAleatorio]

    // remove esse aluno da sua posição na lista
    alunos.splice(indiceAleatorio, 1)

    // adiciona ele no final
    alunos.push(alunoAleatorio) 
}

console.log(`Ordem: ${alunos.join(', ')}`)

File: exs/mundo_1/rust/020.rs
/*
Desafio 020

Problema: O mesmo professor do desafio 019 quer sortear a ordem
          de apresentação de trabalhos dos alunos. Faça um programa
          que leia o nome dos quatro alunos e mostre a ordem sorteada.
*/

use rand::prelude::SliceRandom;

fn main() {
    let mut alunos = Vec::new();

    let mut aluno1 = String::new();
    let mut aluno2 = String::new();
    let mut aluno3 = String::new();
    let mut aluno4 = String::new();

    println!("Digite o nome do aluno 1: ");
    std::io::stdin().read_line(&mut aluno1).unwrap();

    println!("Digite o nome do aluno 2: ");
    std::io::stdin().read_line(&mut aluno2).unwrap();

    println!("Digite o nome do aluno 3: ");
    std::io::stdin().read_line(&mut aluno3).unwrap();

    println!("Digite o nome do aluno 4: ");
    std::io::stdin().read_line(&mut aluno4).unwrap();

    alunos.push(aluno1.trim());
    alunos.push(aluno2.trim());
    alunos.push(aluno3.trim());
    alunos.push(aluno4.trim());

    alunos.shuffle(&mut rand::thread_rng());

    println!("A ordem de apresentação será: {:?}", alunos);
}
